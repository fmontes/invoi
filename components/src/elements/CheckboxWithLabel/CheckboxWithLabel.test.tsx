import * as React from 'react';
import { CheckboxWithLabel } from './index';
import { shallow } from 'enzyme';

test('CheckboxWithLabel should render', () => {
    const component = shallow(<CheckboxWithLabel type="checkbox" />);
    expect(component).toMatchSnapshot();
});

test('CheckboxWithLabel should render disaled', () => {
    const component = shallow(<CheckboxWithLabel disabled type="checkbox" />);
    expect(component).toMatchSnapshot();
});

test('CheckboxWithLabel should render error', () => {
    const component = shallow(<CheckboxWithLabel error type="checkbox" />);
    expect(component).toMatchSnapshot();
});
