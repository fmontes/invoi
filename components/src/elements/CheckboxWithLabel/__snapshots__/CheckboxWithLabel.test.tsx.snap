// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel should render 1`] = `
<Styled(Label)
  error={false}
  theme={
    Object {
      "animation": Object {
        "speed": "100ms",
      },
      "base": Object {
        "radius": Object {
          "normal": "0.5rem",
          "tiny": "0.25rem",
        },
      },
      "colors": Object {
        "background": "#FDFCF9",
        "black": "#202641",
        "disabled": "#D7D0CB",
        "error": "#832400",
        "gray": "#5B6078",
        "neutral": "#CBC1BB",
        "primary": "#3951C5",
        "secondary": "#F0A78B",
        "success": "#3951C5",
        "textBody": "#202641",
        "textSub": "#5B6078",
        "white": "#fff",
      },
      "components": Object {
        "buttons": Object {
          "normal": Object {
            "height": "2.75rem",
          },
          "tiny": Object {
            "height": "1.5rem",
          },
        },
      },
      "font": Object {
        "family": "'Work Sans', sans-serif;",
        "size": Object {
          "body": "1rem",
          "h1": "2rem",
          "h2": "1.5rem",
          "hint": "0.8125rem",
          "tiny": "0.6875rem",
        },
      },
      "spaces": Object {
        "half": "0.25rem",
        "one": "0.5rem",
        "three": "2rem",
        "two": "1rem",
      },
    }
  }
  type="checkbox"
>
  <CheckboxOrRadio
    type="checkbox"
  />
</Styled(Label)>
`;

exports[`CheckboxWithLabel should render disaled 1`] = `
<Styled(Label)
  disabled={true}
  error={false}
  theme={
    Object {
      "animation": Object {
        "speed": "100ms",
      },
      "base": Object {
        "radius": Object {
          "normal": "0.5rem",
          "tiny": "0.25rem",
        },
      },
      "colors": Object {
        "background": "#FDFCF9",
        "black": "#202641",
        "disabled": "#D7D0CB",
        "error": "#832400",
        "gray": "#5B6078",
        "neutral": "#CBC1BB",
        "primary": "#3951C5",
        "secondary": "#F0A78B",
        "success": "#3951C5",
        "textBody": "#202641",
        "textSub": "#5B6078",
        "white": "#fff",
      },
      "components": Object {
        "buttons": Object {
          "normal": Object {
            "height": "2.75rem",
          },
          "tiny": Object {
            "height": "1.5rem",
          },
        },
      },
      "font": Object {
        "family": "'Work Sans', sans-serif;",
        "size": Object {
          "body": "1rem",
          "h1": "2rem",
          "h2": "1.5rem",
          "hint": "0.8125rem",
          "tiny": "0.6875rem",
        },
      },
      "spaces": Object {
        "half": "0.25rem",
        "one": "0.5rem",
        "three": "2rem",
        "two": "1rem",
      },
    }
  }
  type="checkbox"
>
  <CheckboxOrRadio
    disabled={true}
    type="checkbox"
  />
</Styled(Label)>
`;

exports[`CheckboxWithLabel should render error 1`] = `
<Styled(Label)
  error={true}
  theme={
    Object {
      "animation": Object {
        "speed": "100ms",
      },
      "base": Object {
        "radius": Object {
          "normal": "0.5rem",
          "tiny": "0.25rem",
        },
      },
      "colors": Object {
        "background": "#FDFCF9",
        "black": "#202641",
        "disabled": "#D7D0CB",
        "error": "#832400",
        "gray": "#5B6078",
        "neutral": "#CBC1BB",
        "primary": "#3951C5",
        "secondary": "#F0A78B",
        "success": "#3951C5",
        "textBody": "#202641",
        "textSub": "#5B6078",
        "white": "#fff",
      },
      "components": Object {
        "buttons": Object {
          "normal": Object {
            "height": "2.75rem",
          },
          "tiny": Object {
            "height": "1.5rem",
          },
        },
      },
      "font": Object {
        "family": "'Work Sans', sans-serif;",
        "size": Object {
          "body": "1rem",
          "h1": "2rem",
          "h2": "1.5rem",
          "hint": "0.8125rem",
          "tiny": "0.6875rem",
        },
      },
      "spaces": Object {
        "half": "0.25rem",
        "one": "0.5rem",
        "three": "2rem",
        "two": "1rem",
      },
    }
  }
  type="checkbox"
>
  <CheckboxOrRadio
    error={true}
    type="checkbox"
  />
</Styled(Label)>
`;
